compiler/natives/src/net/http/http.go

```diff
@@ -13,18 +13,29 @@ import (
 	"github.com/gopherjs/gopherjs/js"
 )
 
-var DefaultTransport RoundTripper = &XHRTransport{}
+var DefaultTransport = func() RoundTripper {
+	if fetchAPI, streamsAPI := js.Global.Get("fetch"), js.Global.Get("ReadableStream"); fetchAPI != js.Undefined && streamsAPI != js.Undefined {
```

Comment:
It checks that both `fetch` API and `ReadableStream` is available. The former is needed to make Fetch requests, and the latter is to be able to execute this line successfully:

```Go
Body: &streamReader{stream: result.Get("body").Call("getReader")},
```

For example, Firefox currently supports Fetch API but does not support streaming response bodies. So `result.Get("body")` would be nil, and using Fetch over XHR is not very advantageous.